crates/cxx-qt-gen/src/generator/rust/constructor.rs

```diff
@@ -0,0 +1,310 @@
+            parse_quote! {
+                #[namespace = #namespace_internals]
+                #[cxx_name = #arguments_cxx]
+                #[doc(hidden)]
+                struct #arguments_rust {
+                    baseArguments: #base_arguments_rust,
+                    newArguments: #new_arguments_rust,
+                    initializeArguments: #initialize_arguments_rust,
+                }
+            },
+            parse_quote! {
+                #[namespace = #namespace_internals]
+                #[cxx_name = #base_arguments_cxx]
+                #[doc(hidden)]
+                struct #base_arguments_rust {
+                    #(#base_argument_members,)*
+                    notEmpty: i8 // Make sure there's always at least one struct member, as CXX
+                                 // doesn't support empty shared structs.
+                }
+            },
+            parse_quote! {
+                #[namespace = #namespace_internals]
+                #[cxx_name = #new_arguments_cxx]
+                #[doc(hidden)]
+                struct #new_arguments_rust {
+                    #(#new_argument_members,)*
+                    notEmpty: i8 // Make sure there's always at least one struct member, as CXX
+                                 // doesn't support empty shared structs.
+                }
+            },
+            parse_quote! {
+                #[namespace = #namespace_internals]
+                #[cxx_name = #initialize_arguments_cxx]
+                #[doc(hidden)]
+                struct #initialize_arguments_rust {
+                    #(#initialize_argument_members,)*
+                    notEmpty: i8 // Make sure there's always at least one struct member, as CXX
+                                 // doesn't support empty shared structs.
+                }
+            },
+            parse_quote! {
+                extern "Rust" {
+                    #[namespace = #namespace_internals]
+                    #[cxx_name = #route_arguemnts_cxx]
+                    // This function needs to marked unsafe, as some arguments may be pointers.
+                    unsafe fn #route_arguments_rust(#(#route_arguments_parameters),*) -> #arguments_rust;
+
+                    #[namespace = #namespace_internals]
+                    #[cxx_name = #new_cxx]
+                    fn #new_rust(args: #new_arguments_rust) -> Box<#rust_struct_name_rust>;
+
+                    #[namespace = #namespace_internals]
+                    #[cxx_name = #initialize_cxx]
+                    fn #initialize_rust(qobject: Pin<&mut #qobject_name_rust>, args: #initialize_arguments_rust);
+                }
+            },
+        ]);
+        result.cxx_qt_mod_contents.append(&mut vec![parse_quote! {
+            #[doc(hidden)]
+            pub fn #route_arguments_rust(#(#route_arguments_parameters),*) -> #arguments_rust {
+                // these won't be used if any of them don't have any members
```

Comment:
two negatives in this sentence are really hard to read, is it simpler to write it in the opposite way?

Reply:
Agreed, reprhased it.